src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -113,7 +105,28 @@ Status CoreWorkerDirectTaskSubmitter::SubmitTask(TaskSpecification task_spec) {
       RAY_UNUSED(task_finisher_->PendingTaskFailed(
           task_spec.TaskId(), rpc::ErrorType::TASK_CANCELLED, nullptr));
     }
-  });
+  };
+  if (task_spec.IsActorCreationTask()) {
+    // Synchronously register the actor to GCS server.
+    // Previously, we asynchronously registered the actor after all its dependencies were
+    // resolved. This caused a problem: if the owner of the actor dies before dependencies
```

Comment:
```
This caused a problem: if the owner of the actor dies before dependencies
    // are resolved, the actor will never be created. But the actor handle may already be
    // passed to other workers
```

How do we handle this now? Are we not allowing to pass actor handles to other workers until registration is done?

Reply:
Yes, we are not passing it to other workers until it's done. It's considered as a dependence.